public/app/percona/shared/core/reducers/index.ts

```diff
@@ -106,8 +106,14 @@ export const updateSettingsAction = createAsyncThunk(
               args.body.email_alerting_settings!.test_email = undefined;
             }
           }
-          const settings = await SettingsService.setSettings(args.body, args.token);
-          await thunkAPI.dispatch(fetchSettingsAction({ usedPassword: password, testEmail }));
+          let settings;
```

Comment:
I don't it's needed to refetch the settings if the update failed. I think you can let the `withAppEvents` catch the initial error.